lib/vagrant-azure/monkey_patch/winrm.rb

```diff
@@ -0,0 +1,77 @@
+# patch no_ssl_peer_validation for ssl based winrm
+
+WinRM::HTTP::HttpTransport.class_eval do
+
+  # provide a patch for no peer verification.
+  # the patch exists upstream, but vagrant depends on an older version of winrm
```

Comment:
The no_ssl_peer_verification was introduced in winrm here https://github.com/WinRb/WinRM/commit/f4ece4384df2768bc7756c3c5c336db65b05c674 and is available in 1.2.0 and greater. The Vagrant embedded version of that gem is 1.1.3.